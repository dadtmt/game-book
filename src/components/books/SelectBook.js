import React from 'react'
import { Link } from 'react-router-dom'

const SelectBook = ({id, description, imageUrl, title}) => <div>
  <Link to={`book/${id}`}>Select this book link</Link>
  <h2>{title}</h2>
  { imageUrl && (
    <div
      style={{
        backgroundImage: `url(${imageUrl})`,
        backgroundSize: 'cover',
        paddingBottom: '100%',
      }}
    />
  ) }
  <div>
    {description}&nbsp;
  </div>
</div>

export default SelectBook
