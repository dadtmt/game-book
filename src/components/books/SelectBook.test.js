import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import SelectBook from './SelectBook'

describe('SelectBook component', () => {
  it('component renders correctly', () => {
    const props = {
      id: 'ID',
      description: `Verum ad istam omnem orationem brevis est defensio. Nam quoad aetas M. Caeli dare potuit isti suspicioni locum, fuit primum ipsius pudore, deinde etiam patris diligentia disciplinaque munita. Qui ut huic virilem togam dedit≈°nihil dicam hoc loco de me; tantum sit, quantum vos existimatis; hoc dicam, hunc a patre continuo ad me esse deductum; nemo hunc M. Caelium in illo aetatis flore vidit nisi aut cum patre aut mecum aut in M. Crassi castissima domo, cum artibus honestissimis erudiretur.`,
      imageUrl: 'http://lorempixel.com/400/200/',
      title: 'Book Title'
    }
    const tree = renderer
      .create(
        <BrowserRouter>
          <SelectBook {...props} />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
