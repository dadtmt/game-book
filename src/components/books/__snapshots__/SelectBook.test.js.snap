// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBook component component renders correctly 1`] = `
<div>
  <a
    href="book/ID"
    onClick={[Function]}
  >
    Select this book link
  </a>
  <h2>
    Book Title
  </h2>
  <div
    style={
      Object {
        "backgroundImage": "url(http://lorempixel.com/400/200/)",
        "backgroundSize": "cover",
        "paddingBottom": "100%",
      }
    }
  />
  <div>
    Verum ad istam omnem orationem brevis est defensio. Nam quoad aetas M. Caeli dare potuit isti suspicioni locum, fuit primum ipsius pudore, deinde etiam patris diligentia disciplinaque munita. Qui ut huic virilem togam deditšnihil dicam hoc loco de me; tantum sit, quantum vos existimatis; hoc dicam, hunc a patre continuo ad me esse deductum; nemo hunc M. Caelium in illo aetatis flore vidit nisi aut cum patre aut mecum aut in M. Crassi castissima domo, cum artibus honestissimis erudiretur.
     
  </div>
</div>
`;
